driver-core/src/main/com/mongodb/client/gridfs/codecs/GridFSFileCodec.java

```diff
@@ -40,18 +41,24 @@
 import static java.util.Arrays.asList;
 
 
-final class GridFSFileCodec implements Codec<GridFSFile> {
-    private static final BsonDocumentCodec BSON_DOCUMENT_CODEC = new BsonDocumentCodec();
+/**
+ * A codec for GridFS Files
+ *
+ * @since 3.3
+ */
+public final class GridFSFileCodec implements Codec<GridFSFile> {
     private static final List<String> VALID_FIELDS = asList("_id", "filename", "length", "chunkSize", "uploadDate", "md5", "metadata");
     private final Codec<Document> documentCodec;
+    private final Codec<BsonDocument> bsonDocumentCodec;
 
     public GridFSFileCodec(final CodecRegistry registry) {
         this.documentCodec = notNull("DocumentCodec", notNull("registry", registry).get(Document.class));
+        this.bsonDocumentCodec = notNull("BsonDocumentCodec", registry.get(BsonDocument.class));
```

Comment:
By contract this can't be null, as CodecRegistry#get will throw rather than return null.